ios/Exponent/Versioned/Core/UniversalModules/EXNotifications/EXScopedNotificationsUtils.m

```diff
@@ -18,4 +18,71 @@ + (BOOL)shouldNotification:(UNNotification *)notification beHandledByExperience:
+  NSString *escapedCategoryId = [EXScopedNotificationsUtils escapedString:categoryId];
+  return [NSString stringWithFormat:@"%@/%@", scope, escapedCategoryId];
+}
+
++ (BOOL)isCategoryId:(NSString *)identifier scopedByExperience:(NSString *)experienceId
+{
+  NSString *scopeFromCategoryId = [EXScopedNotificationsUtils getScopeAndIdentifierFromScopedIdentifier:identifier].scopeKey;
+  return [scopeFromCategoryId isEqualToString:experienceId];
+}
+
++ (ScopedCategoryIdentifierComponents)getScopeAndIdentifierFromScopedIdentifier:(NSString *)scopedIdentifier
+{
+  NSString *scope = @"";
+  NSString *identifier = @"";
+  NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"^(.*(?<!\\\\)|.*(?:\\\\\\\\)+)/(.*)$"
+                                                                         options:0
+                                                                           error:nil];
+  NSTextCheckingResult *match = [regex firstMatchInString:scopedIdentifier
+                                                  options:0
+                                                    range:NSMakeRange(0, scopedIdentifier.length)];
+  if (!match) {
+    // No delimiter found, so no scope associated with this identifier
+    identifier = scopedIdentifier;
+  } else {
+    scope = [scopedIdentifier substringWithRange:[match rangeAtIndex:1]];
+    identifier = [scopedIdentifier substringWithRange:[match rangeAtIndex:2]];
+  }
+  ScopedCategoryIdentifierComponents components;
+  components.scopeKey = [EXScopedNotificationsUtils unescapedString:scope];
+  components.categoryIdentifier = [EXScopedNotificationsUtils unescapedString:identifier];
+  return components;
+}
+
++ (NSString *)escapedString:(NSString*)string
+{
+  return [[string stringByReplacingOccurrencesOfString:@"\\" withString:@"\\\\"]
+    stringByReplacingOccurrencesOfString:@"/" withString:@"\\/"];
+}
+
++ (NSString *)unescapedString:(NSString*)string
+{
+  return [[string stringByReplacingOccurrencesOfString:@"\\/" withString:@"/"]
+          stringByReplacingOccurrencesOfString:@"\\\\" withString:@"\\"];
+}
+
+# pragma mark Legacy notification category scoping
+
++ (BOOL)isLegacyCategoryId:(NSString *) scopedCategoryId scopedByExperience:(NSString *) experienceId
+{
+  NSString* legacyScopingPrefix = [NSString stringWithFormat:@"%@-", experienceId];
+  return [scopedCategoryId hasPrefix:legacyScopingPrefix];
+}
+
+// legacy categories were stored under an unescaped experienceId
++ (NSString *)unscopedLegacyCategoryIdentifierWithId:(NSString *) scopedCategoryId
+                                       forExperience:(NSString *) experienceId
+{
+  NSString* legacyScopingPrefix = [NSString stringWithFormat:@"%@-", experienceId];
+  return [scopedCategoryId stringByReplacingOccurrencesOfString:legacyScopingPrefix
+                                                     withString:@""
+                                                        options:NSAnchoredSearch
```

Comment:
`options:NSAnchoredSearch` means we will only match if it's a prefix. Covers our bases if users happen to have the legacy scope _in_ their own category ID, i.e.:
```
@charliecruzan/mySlug-categoryNameFor@charliecruzan/mySlug-`
```